webui/react/src/hooks/useModal/Checkpoint/useModalCheckpointDelete.tsx

```diff
@@ -0,0 +1,80 @@
+import { ExclamationCircleOutlined } from '@ant-design/icons';
+import { ModalFuncProps } from 'antd';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+
+import { detApi } from 'services/apiConfig';
+import { readStream } from 'services/utils';
+import useModal, { ModalHooks as Hooks, ModalCloseReason } from 'shared/hooks/useModal/useModal';
+import { pluralizer } from 'shared/utils/string';
+
+interface OpenProps {
+  checkpoints: string | string[];
+  initialModalProps?: ModalFuncProps;
+}
+
+export interface Props {
+  onClose?: (reason?: ModalCloseReason) => void;
+}
+
+interface ModalHooks extends Omit<Hooks, 'modalOpen'> {
+  modalOpen: (openProps?: OpenProps) => void;
+}
+
+const useModalCheckpointDelete = ({ onClose }: Props): ModalHooks => {
+  const { modalOpen: openOrUpdate, modalRef, ...modalHook } = useModal();
+  const [ checkpoints, setCheckpoints ] = useState<string | string[]>([]);
+
+  const numCheckpoints = useMemo(() => {
+    if (Array.isArray(checkpoints)) return checkpoints.length;
+    return 1;
+  }, [ checkpoints ]);
+
+  const handleCancel = useCallback(() => onClose?.(ModalCloseReason.Cancel), [ onClose ]);
+
+  const handleDelete = useCallback(() => {
+    readStream(detApi.Checkpoint.deleteCheckpoints({
+      checkpointUuids: Array.isArray(checkpoints) ?
+        checkpoints :
+        [ checkpoints ],
+    }));
+    onClose?.(ModalCloseReason.Ok);
+  }, [ checkpoints, onClose ]);
+
+  const modalProps: ModalFuncProps = useMemo(() => {
+    const content =
+      `Are you sure you want to request checkpoint deletion for all batches in
```

Comment:
I think a checkpoint always corresponds to a single batch.

The text in the existing useModalCheckpoint file should read "checkpoint deletion for batch ${totalBatches}"